decidim-admin/app/commands/decidim/admin/create_component.rb

```diff
@@ -3,50 +3,16 @@
 module Decidim
   module Admin
     # This command gets called when a component is created from the admin panel.
-    class CreateComponent < Decidim::Command
-      attr_reader :form, :manifest, :participatory_space
-
-      # Public: Initializes the command.
-      #
-      # form - The form from which the data in this component comes from.
-      def initialize(form)
-        @form = form
-        @manifest = form.manifest
-      end
-
-      # Public: Creates the Component.
-      #
-      # Broadcasts :ok if created, :invalid otherwise.
-      def call
-        return broadcast(:invalid) if form.invalid?
-
-        transaction do
-          create_component
-          run_hooks
-        end
-
-        broadcast(:ok)
-      end
+    class CreateComponent < Decidim::Commands::CreateResource
+      fetch_form_attributes :name, :participatory_space, :weight, :settings, :default_step_settings, :step_settings
 
       private
 
-      def create_component
-        @component = Decidim.traceability.create!(
-          Component,
-          form.current_user,
-          manifest_name: manifest.name,
-          name: form.name,
-          participatory_space: form.participatory_space,
-          weight: form.weight,
-          settings: form.settings,
-          default_step_settings: form.default_step_settings,
-          step_settings: form.step_settings
-        )
-      end
+      def attributes = super.reverse_merge({ manifest_name: form.manifest.name })
```

Comment:
```suggestion
      # Use `reverse_merge` instead of `merge` as we need the `manifest_name` first
      def attributes = super.reverse_merge({ manifest_name: form.manifest.name })
```